### PR TITLE
chore: push commits from personal repo

### PR DESCRIPTION
**Pull request by vjcagay**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+			gulp.src( paths.test, { 'read' : false } )
+				.pipe( mocha( mochaOpts )
+					.on( 'error', function ( err ) {
+						throw new Error( err );
+					} ) )
+				.pipe( istanbul.writeReports( {
+					'reporters' : [ 'text', 'text-summary' ]
+				} ) )
+				.pipe( istanbul.enforceThresholds( {
```
Comment (fakewaffle) [inline gulpfile.js]: Please use https://github.com/School-Improvement-Network/coverage-enforcement/blob/master/package.json#L2.

Check https://github.com/School-Improvement-Network/legacy-service/blob/dev/gulpfile.js for implementation.

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		.pipe( istanbul.hookRequire() )
+		.on( 'finish', function () {
+
+			gulp.src( paths.test, { 'read' : false } )
+				.pipe( mocha( mochaOpts )
+					.on( 'error', function ( err ) {
+						throw new Error( err );
+					} ) )
+				.pipe( istanbul.writeReports( {
+					'reporters' : [ 'text', 'text-summary' ]
```
Comment (fakewaffle) [inline gulpfile.js]: Check https://github.com/School-Improvement-Network/legacy-service/blob/dev/gulpfile.js for implementation.

**Comment by fakewaffle** [comment]
Update `gulpfile.js` to be consistent with https://github.com/School-Improvement-Network/legacy-service/blob/dev/gulpfile.js.

File: gulpfile.js (hunk 1 above)
Comment (vjcagay) [inline gulpfile.js]: I put a global coverage setting of 100% on everything as a requirement. Should I still change it to sinet's settings?

File: README.md [hunk 3]
```diff
@@ -1 +1,102 @@
...
+- `removeFileExt` file extensions are ignored in checking when set to `true`. This is useful when you have a custom `RegExp` for a set of files and you decide to include the file extensions in checking. Defaults to `true` when not explicitly set.
+
+> *Warning:*
+> If you have `foo.test.js` with `removeFileExt` enabled, `.js` will be ignored and only `foo.test` will be checked.
+
+- `globRegexp` accepts a set of globs and their corresponding `RegExp` patterns. Patterns could be one of the presets or custom RegExp object you can define. In the example above, `'test/*.js' : 'HYPHEN_CASE'` means all files that have the `.js` file extension will have their filenames checked whether they satify the `HYPHEN_CASE` pattern or not.
+
+> Read more [here](https://github.com/isaacs/node-glob) to know more how globbing works.
+> Presets assume `removeFileExt` is `true`. You need to create your own `RegExp` pattern if you want file extensions to be included in checking. To see the list of presets, go [here](https://github.com/vjcagay/filedirname-lint/blob/master/lib/presets.js).
```
Comment (openam) [inline README.md]: update link location.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
+{
+  "name": "@sinet/pathlint",
+  "version": "1.0.0",
+  "description": "Enforces specific file/directory name patterns for uniformity. It matches files/directories in the current working directory. Great for projects involving several developers.",
+  "author": {
+    "name": "Vincent John Cagay",
```
Comment (openam) [inline package.json]: Should we have the author be `School Improvement Network` to be consistent with the other open source projects we've released, e.g. https://github.com/School-Improvement-Network/lapin/blob/master/package.json#L16

File: package.json [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    "type": "git",
+    "url": "https://github.com/School-Improvement-Network/pathlint.git"
+  },
+  "bugs": {
+    "url": "https://github.com/School-Improvement-Network/pathlint/issues"
+  },
+  "license": "MIT",
+  "main": "./lib/pathlint.js",
+  "scripts": {
+    "test": "npm run eslint -- --reset --eslintrc ./ && npm run jscs -- ./ && npm run pathlint && gulp test",
```
Comment (openam) [inline package.json]: Should test run the linting stuff, or should it be separated out? Test seems like it should just run tests to me. On the flip side maybe we should add a convenience `npm run lint` that would run all the linting (eslint, jscs, pathlint).

File: README.md (hunk 3 above)
Comment (vjcagay) [inline README.md]: Sorry, will update this.

File: package.json (hunk 4 above)
Comment (vjcagay) [inline package.json]: Sorry, will update this one. Thanks for noting.

File: package.json (hunk 5 above)
Comment (fakewaffle) [inline package.json]: Test should run tests. Linting should be separate.

File: gulpfile.js (hunk 1 above)
Comment (fakewaffle) [inline gulpfile.js]: Update gulpfile.js to be consistent with https://github.com/School-Improvement-Network/legacy-service/blob/dev/gulpfile.js.

File: package.json (hunk 5 above)
Comment (vjcagay) [inline package.json]: I think separating the unit tests and linting will be beneficiary in the long run, especially when projects get larger as more modules are added. We could do that right now.

File: gulpfile.js (hunk 1 above)
Comment (vjcagay) [inline gulpfile.js]: Updated gulpfile.js

File: README.md (hunk 3 above)
Comment (vjcagay) [inline README.md]: link updated.

File: package.json (hunk 4 above)
Comment (vjcagay) [inline package.json]: author updated.

File: package.json (hunk 5 above)
Comment (vjcagay) [inline package.json]: I separated linting and unit testing.

**Comment by fakewaffle** [comment]
@gideonairex can you merge this if it's GTG?

**Comment by gideonairex** [comment]
@fakewaffle okay I will check this back.

**Pull request merged**